ansible/configs/multi-region-tower/deploy_stack.yml

```diff
@@ -0,0 +1,77 @@
+---
+- debug:
+    msg: "stack_file is {{stack_file}}"
+- when: stack_file == 'default'
+  block:
+    - name: Run infra-ec2-template-generate Role (target_regions)
+      include_role:
+        name: infra-ec2-template-generate
+      vars:
+        cloudformation_template: "{{ output_dir }}/{{ env_type }}.{{ guid }}.{{ cloud_provider }}_cloud_template_{{ aws_region }}"
+
+    - set_fact:
+        stack_deployed: false
+    - name: Run infra-ec2-template-create Role (target_regions)
+      include_role:
+        name: infra-ec2-template-create
+      vars:
+        aws_region_loop: "{{ aws_region }}"
+        cloudformation_template: "{{ output_dir }}/{{ env_type }}.{{ guid }}.{{ cloud_provider }}_cloud_template_{{ aws_region }}"
+
+- when: stack_file != 'default'
+  block:
+    - name: Run infra-ec2-template-generate Role (target_regions)
+      include_role:
+        name: infra-ec2-template-generate
+      vars:
+        cloudformation_template_src: "../../configs/{{ env_type }}/files/cloud_providers/{{ stack_file }}"
+        cloudformation_template: "{{ output_dir }}/{{ env_type }}.{{ guid }}.{{ cloud_provider }}_cloud_template_{{ aws_region }}"
+ 
+    - set_fact:
+        stack_deployed: false
+
+    - name: Run infra-ec2-template-create Role (target_regions)
+      include_role:
+        name: infra-ec2-template-create
+      vars:
+        aws_region_loop: "{{ aws_region }}"
+        cloudformation_template: "{{ output_dir }}/{{ env_type }}.{{ guid }}.{{ cloud_provider }}_cloud_template_{{ aws_region }}"
```

Comment:
It seems we don't need this test:

```yaml
- when: stack_file == 'default'
```
and 

```yaml
- when: stack_file != 'default'
```

The 2 blocks being identical we can just merge them and remove the test.